R/indexing.R

```diff
@@ -53,6 +53,66 @@
 # - G[1:3,2,eid=TRUE]
 #               create an edge sequence
 
+get_adjacency_submatrix <- function(x, i = NULL, j = NULL, attr = NULL, sparse = TRUE) {
```

Comment:
As agreed: this should be 

```suggestion
get_adjacency_submatrix_unique <- function(x, i = NULL, j = NULL, attr = NULL, sparse = TRUE) {
```

with the precondition that `i` and `j` are unique (or `NULL`), and we add a new `get_adjacency_submatrix()` that checks `anyDuplicated(...)` and blows up the resulting matrix with

```r
unique_res[i_map, j_map, drop = FALSE]
```

but only if needed.